tasks/update_go.py

```diff
@@ -0,0 +1,179 @@
+import re
+from typing import Optional
+
+from invoke import Context, exceptions, task
+
+from .go import tidy_all
+from .libs.common.color import color_message
+from .modules import DEFAULT_MODULES
+from .pipeline import update_circleci_config, update_gitlab_config
+
+GO_VERSION_FILE = "./.go-version"
+
+
+# replace the given pattern with the given string in the file
+def _update_file(path: str, pattern: str, replace: str, expected_match: int = 1):
+    # newline='' keeps the file's newline character(s)
+    # meaning it keeps '\n' for most files and '\r\n' for windows specific files
+
+    with open(path, "r", newline='') as reader:
+        content = reader.read()
+
+    content, nb_match = re.subn(pattern, replace, content, flags=re.MULTILINE)
+    if nb_match != expected_match:
+        print(
+            color_message(
+                f"WARNING: {path}: '{pattern}': expected {expected_match} matches but go {nb_match}", "orange"
+            )
+        )
+
+    with open(path, "w", newline='') as writer:
+        writer.write(content)
+
+
+# returns the current go version
+def _get_repo_go_version() -> str:
+    with open(GO_VERSION_FILE, "r") as reader:
+        version = reader.read()
+    return version.strip()
+
+
+# extracts the major version from the given string
+# eg. if the string is "1.2.3", returns "1.2"
+def _get_major_version(version: str) -> str:
```

Comment:
Note: You can use something like https://pypi.org/project/semver/ to parse the version. It is also fine to use `rpatition`

Reply:
Oh that's nice, I was thinking it would be nice to also support using release candidates of `go` versions (if we ever want to try) but that it would be annoying to parse all possible version syntax :)